helm-charts/chatqna/templates/deployment.yaml

```diff
@@ -33,6 +33,13 @@ spec:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
       containers:
         - name: {{ .Release.Name }}
+          {{- if and .Values.externalLLM (eq .Values.externalLLM.enabled true) }}
```

Comment:
just `{{- if .Values.externalLLM.enabled }}` should be enough if we have set default values under `externalLLM` key in values.yaml